### PR TITLE
services: socket options were erroneously ignored if socket has no stats

### PR DESCRIPTION
**Pull request by zpencer**
This fixes listen sockets. It is ok to have no data but report socket
options.

### CODE REVIEW

File: core/src/main/java/io/grpc/internal/Channelz.java [hunk 1]
```diff
@@ -468,9 +468,9 @@ public SocketStats(
         SocketOptions socketOptions,
         Security security) {
       this.data = data;
-      this.local = local;
+      this.local = Preconditions.checkNotNull(local);
```
Comment (carl-mastrangelo) [inline core/src/main/java/io/grpc/internal/Channelz.java]: consider adding the variable name as a second arg

File: okhttp/src/test/java/io/grpc/okhttp/OkHttpClientTransportTest.java [hunk 2]
```diff
@@ -1982,6 +1983,11 @@ static String getContent(InputStream message) {
     public synchronized void close() {
       frameReader.nextFrameAtEndOfStream();
     }
+
+    @Override
+    public SocketAddress getLocalSocketAddress() {
+      return InetSocketAddress.createUnresolved("127.0.0.1", 4000);
```
Comment (carl-mastrangelo) [inline okhttp/src/test/java/io/grpc/okhttp/OkHttpClientTransportTest.java]: localhost?

File: services/src/main/java/io/grpc/services/ChannelzProtoUtil.java [hunk 3]
```diff
@@ -168,29 +166,31 @@ static Address toAddress(SocketAddress address) {
...
+          .setLastLocalStreamCreatedTimestamp(
+              Timestamps.fromNanos(s.lastLocalStreamCreatedTimeNanos))
+          .setLastRemoteStreamCreatedTimestamp(
+              Timestamps.fromNanos(s.lastRemoteStreamCreatedTimeNanos))
+          .setLastMessageSentTimestamp(
+              Timestamps.fromNanos(s.lastMessageSentTimeNanos))
+          .setLastMessageReceivedTimestamp(
+              Timestamps.fromNanos(s.lastMessageReceivedTimeNanos))
+          .setLocalFlowControlWindow(
+              Int64Value.newBuilder().setValue(s.localFlowControlWindow).build())
```
Comment (carl-mastrangelo) [inline services/src/main/java/io/grpc/services/ChannelzProtoUtil.java]: The interior builds() are unnecessary I think

File: services/src/test/java/io/grpc/services/ChannelzProtoUtilTest.java [hunk 4]
```diff
@@ -333,14 +334,55 @@ public void toState() {
...
+            .newBuilder()
+            .setRef(socketRef)
+            .setLocal(localAddress)
+            .setRemote(remoteAddress)
+            .setData(
+                SocketData
+                    .newBuilder(socketDataWithDataNoSockOpts)
+                    .addOption(
+                        SocketOption.newBuilder()
+                            .setName("test_name").setValue("test_value").build())
```
Comment (carl-mastrangelo) [inline services/src/test/java/io/grpc/services/ChannelzProtoUtilTest.java]: same here.

File: core/src/main/java/io/grpc/internal/Channelz.java (hunk 1 above)
Comment (zpencer) [reply core/src/main/java/io/grpc/internal/Channelz.java]: Added description

File: okhttp/src/test/java/io/grpc/okhttp/OkHttpClientTransportTest.java (hunk 2 above)
Comment (zpencer) [reply okhttp/src/test/java/io/grpc/okhttp/OkHttpClientTransportTest.java]: changed to localhost

File: services/src/main/java/io/grpc/services/ChannelzProtoUtil.java (hunk 3 above)
Comment (zpencer) [reply services/src/main/java/io/grpc/services/ChannelzProtoUtil.java]: Changed to `.of()`

File: services/src/test/java/io/grpc/services/ChannelzProtoUtilTest.java (hunk 4 above)
Comment (zpencer) [reply services/src/test/java/io/grpc/services/ChannelzProtoUtilTest.java]: Removed this and other internal builds

**Pull request merged**